web/client/components/widgets/builder/wizard/ChartWizard.jsx

```diff
@@ -139,6 +139,11 @@ const ChartWizard = ({
             data
         };
     });
+    useEffect(() => {
+        if (selectedTrace?.type === "pie" && tab === "axis") {
+            setTab("traces");
+        }
+    }, [selectedTrace?.type]);
```

Comment:
I don't think a useEffect is needed in this case because the current selected tab could be computed directly in the render:

```js
const selectedTab = selectedTrace?.type === "pie" && tab === "axis" ? "traces" : (selectedTrace?.type || "traces");
```

then use:

```js
{tabContents[selectedTab]}
```

the selectedTab can be also passed to other components as tab prop

Reply:
are you sure about this line, especially in the false part selectedTab could be = to selectedTrace?.type which is not a tab

![image](https://github.com/geosolutions-it/MapStore2/assets/11991428/3376cdb6-27ba-461a-a806-0dd4ad19da06)

maybe it was `tab || "traces"` or simply tab?

Reply:
yes, it was like that, I wrote the code in the comment not tested